script.js

```diff
@@ -0,0 +1,364 @@
+initializeCalculator();
+function initializeCalculator() {
+  let first = {value: "", unsaved: true, saved: false, integer: true};
+  let second = {value: "", unsaved: true, saved: false, integer: true};
+  let operator = {value: "", unsaved: true, saved: false};
+  let result = "";
+  let allowNewNumber = false;
+
+  listenClicks();
+  listenKeys();
+
+  function listenKeys() {
+    window.addEventListener("keydown", event => {
+      const key = document.querySelector(`[data-key="${event.key}"]`);
+      if (key) {
+        bindEvent(key.id);
+        key.focus(); // Safari and Firefox don't handle focus in CSS files
+      }
+    });
+  }
+
+  function listenClicks() {
+    const buttons = document.querySelectorAll(".btn");
+    buttons.forEach(button => {
+      button.addEventListener("click", event => {
+        const click = event.target;
+        bindEvent(click.id);
+        click.focus(); // Safari and Firefox don't handle focus in CSS files
+      });
+    });
+  }
+
+  function bindEvent(event) {
+    if (["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"].includes(event)) {
+      addOperand(event);
+    }
+    if (["+", "-", "÷", "×", "^"].includes(event)) {
+      addOperator(event);
+    }
+    switch (event) {
+      case "=":
+        evaluateExpression(first, operator, second);
+        break;
+      case ".":
+        addDecimal(event);
+        break;
+      case "+/-":
+        getOpposite();
+        break;
+      case "%":
+        getPercentage();
+        break;
+      case "ac":
+        clearExpression();
+        break;
+    }
+    updateClearbutton("AC", "C");
+    updateDisplay(first.value, operator.value, second.value, result);
+    console.table({
+      first: first,
+      operator: operator,
+      second: second,
+      result: result,
+      new: allowNewNumber
+    });
+  }
+
+  function addOperand(operandEvent) {
+    // Handle the other digits for the second number
+    if (first.saved && operator.saved && second.saved) {
+      second.value += operandEvent;
+      // Handle the leading zero for the second number
+      if (second.value.startsWith("0") && second.value.slice(1, 2) !== ".") {
```

Comment:
These if/else conditions are very hard to read. I think you should create a short function with a descriptive name explaining what `second.value.startsWith("0") && second.value.slice(1, 2) !== "."` means instead of writing the comment `// Handle the leading zero for the second number`

Reply:
The equivalent is true for all other if/else conditions.